src/orderbook/orderbooks.py

```diff
@@ -33,53 +34,33 @@ class SimpleOrderBook:
     def __init__(self, bid: Stack, ask: Stack, kafka_client: KafkaClient):
         self.bid = bid
         self.ask = ask
+        self.lock = threading.Lock()
```

Comment:
If we thread lock at init it makes the object not sharable. I think only the matching need to be thread locking